### ISSUE TITLE
Implement the training loop: Forward pass (Render) -> Loss -> Backward pass -> Optimizer Step

### ISSUE THREAD

**Issue by hugoschreib** [JulesRD/gaussian-splatting-volumetrics#10]
FIle: src/training/trainer.py
